operator/controllers/mlops/server_controller.go

```diff
@@ -142,13 +135,23 @@ func (r *ServerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return reconcile.Result{}, err
 	}
 
+	// attempt to deploy
 	err = sr.Reconcile()
 	if err != nil {
 		logger.Error(err, "Failed reconciling", "name", req.Name, "namespace", req.Namespace, "spec", server.Spec)
 		r.updateStatusFromError(ctx, logger, server, err)
 		return reconcile.Result{}, err
 	}
 
+	// only after successful deployment of the pods, we notify the scheduler of the number of Servers it can expect that
+	// will try and connect. It's important we do this after we deploy, otherwise if deployment fails, the scheduler
+	// may expect the wrong number, and end up waiting for the 10-minute timeout until loading models onto the Servers.
+	if err := r.Scheduler.ServerNotify(ctx, nil, []mlopsv1alpha1.Server{*server}, false); err != nil {
```

Comment:
The thing that I'm unconvinced about here is that the initial scheduler sync is not done via _this_ `ServerNotify` (at least not directly).

Rather, it is done via the `ServerNotify` sent as a response to `ControlPlaneResponse_SEND_SERVERS`, [here](https://github.com/SeldonIO/seldon-core/blob/a83ff7bcd2be5f69a3eb6f792e9e483190e0b805/operator/scheduler/utils.go#L139). That one still simply lists the Server CRs, and the scheduler, when receiving the response from the operator, will count their expected replicas [here](https://github.com/SeldonIO/seldon-core/blob/a83ff7bcd2be5f69a3eb6f792e9e483190e0b805/scheduler/pkg/server/server.go#L434)